### ISSUE TITLE
[UI Extensions] Missing types for weightType and weightGroup

### ISSUE THREAD

**Issue by mqufflc** [rancher/dashboard#13858]
**Setup**
- Rancher version: 2.10-head
- Rancher UI Extensions: N/A
- Browser type & version: N/A
- shell version: 3.0.2

**Describe the bug**

When following the getting started guide, the `weightType` and `weightGroup` functions are presented in order to order the pages link in the side bar when developping an extention [link to docs](https://extensions.rancher.io/extensions/next/api/nav/side-menu#side-menu-ordering-weighttype-and-weightgroup).

In fact the following code sample doesn't import the functions from `$plugin.DSL` (but it is stated).

When trying to do so locally, my editor warns me that the functions are not exported by `$plugin.DSL`:
`Property 'weightType' does not exist on type 'DSLReturnType'`.

Indeed, it seems like thes two functions were not included in the interface `DSLReturnType` of the @rancher/shell package: [permalink](https://github.com/rancher/dashboard/blob/d40cacdd3f9a546affa94a506cd6db6789f64880/shell/core/types.ts#L487)

No issue for building the dev server nor for building the extension.

**To Reproduce**

Try to follow the getting started for using weightType and weightGroup functions for inclusing pages in a rancher extension.

**Result**

There will be no autocompletion and the editor will complain that these properties doesn't exists.

**Expected Result**

The functions are definde in the interface DSLReturnType.

**Screenshots**

![Image](https://github.com/user-attachments/assets/5de35ed8-7152-49c9-91ed-a7b80ca2240c)

**Additional context**

N/A
